### PR TITLE
Adapt interestPerBlock calc to recreate 2.5.0 conditions

### PR DESCRIPTION
**Pull request by prasannavl**
- Adapts interest calculation to match 2.5.0 conditions, even though 2.6.0 calculations are the correct ones. (Missed bug from 2.5.0). 
- Add FCH guard for swap. 
- Minor maintainer friendly renames.

### CODE REVIEW

File: src/masternodes/mn_checks.cpp [hunk 1]
```diff
@@ -2834,9 +2836,9 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
                 tokenId = tokenDUSD->first;
 
                 // Calculate the DFI amount in DUSD
-                paybackAmount = MultiplyAmounts(dfiUSDPrice, kv.second);
-                if (dfiUSDPrice > COIN && paybackAmount < kv.second)
-                    return Res::Err("Value/price too high (%s/%s)", GetDecimaleString(kv.second), GetDecimaleString(dfiUSDPrice));
+                paybackAmount = MultiplyAmounts(dfiUSDPrice, paybackAmount);
```
Comment (kuegi) [inline src/masternodes/mn_checks.cpp:2839]: wouldn't it be clearer to read to make this `paybackAmount = MultiplyAmounts(dfiUSDPrice, tokenLoanBalance);`
Specially with the comparison in the next line?

File: src/masternodes/mn_checks.cpp [hunk 2]
```diff
@@ -2922,15 +2924,15 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
...
                 else
                 {
-                    subInDFI = kv.second;
+                    subInDFI = tokenLoanBalance;
                 }
 
                 CDataStructureV0 liveKey{AttributeTypes::Live, ParamIDs::Economy, EconomyKeys::PaybackDFITokens};
                 auto balances = attributes->GetValue(liveKey, CBalances{});
-                auto penaltyDFI = MultiplyAmounts(subInDFI, COIN - penaltyPct);
+                auto tokenEquivalentDFI = MultiplyAmounts(subInDFI, COIN - penaltyPct);
```
Comment (kuegi) [inline src/masternodes/mn_checks.cpp:2932]: but this is not the tokenEquivalent. This is only the penalty (the 1% currently). or am I mistaken here? (penaltyPct = 99% currently)

Comment (prasannavl) [reply src/masternodes/mn_checks.cpp:2932]: Please ignore. Superceded by 1089

File: src/masternodes/mn_checks.cpp (hunk 1 above)
Comment (prasannavl) [reply src/masternodes/mn_checks.cpp:2839]: Please ignore. Superceded by 1089

**Pull request closed without merging**